include/nbl/builtin/hlsl/workgroup2/arithmetic_config.hlsl

```diff
@@ -71,16 +53,32 @@ struct ArithmeticConfiguration
     NBL_CONSTEXPR_STATIC_INLINE uint16_t ItemsPerInvocation_2 = items_per_invoc_t::value2;
     static_assert(ItemsPerInvocation_1<=4, "3 level scan would have been needed with this config!");
 
-    NBL_CONSTEXPR_STATIC_INLINE uint16_t ElementCount = conditional_value<LevelCount==1,uint16_t,0,conditional_value<LevelCount==3,uint16_t,SubgroupSize*ItemsPerInvocation_2,0>::value + SubgroupSize*ItemsPerInvocation_1>::value;
+    NBL_CONSTEXPR_STATIC_INLINE uint32_t SharedScratchElementCount = conditional_value<LevelCount==1,uint16_t,
+        0,
+        conditional_value<LevelCount==3,uint16_t,
+            SubgroupSize*ItemsPerInvocation_2,
+            0
+            >::value + SubgroupSize*ItemsPerInvocation_1
+        >::value;
+
+    static bool electLast()
+    {
+        return glsl::gl_SubgroupInvocationID()==SubgroupSize-1;
+    }
+
+    static uint32_t virtualSubgroupID(const uint32_t subgroupID, const uint32_t virtualIdx)
+    {
+        return virtualIdx * (WorkgroupSize >> SubgroupSizeLog2) + subgroupID;
+    }
 
-    static uint32_t virtualSubgroupID(const uint32_t id, const uint32_t offset)
+    static uint32_t sharedCoalescedIndexNextLevel(const uint32_t subgroupID, const uint32_t itemsPerInvocation)
     {
-        return offset * (WorkgroupSize >> SubgroupSizeLog2) + id;
+        return (subgroupID & (itemsPerInvocation-1)) * SubgroupSize + (subgroupID/itemsPerInvocation);
```

Comment:
its still wrong due to the `SubgroupSize` multiplication and your `itemsPerInovcation` is still in the Config and shouldn't be passed from outside